databus/src/main/java/com/bazaarvoice/emodb/databus/core/DefaultDatabus.java

```diff
@@ -476,91 +477,109 @@ private PollResult peekOrPoll(String subscription, @Nullable Duration claimTtl,
+                            eventIdsToDiscard.add(eventId);
+                            _redundantMeter.mark();
+                            continue;
+                        }
 
-                    // Check whether we've already added this piece of content to the poll result.  If so, consolidate
-                    // the two together to reduce the amount of work a client must do.  Note that the previous item
-                    // might be from a previous batch of events and it's possible that we have read two different
-                    // versions of the same item of content.  This will prefer the most recent.
-                    Item previousItem = uniqueItems.get(coord);
-                    if (previousItem != null && previousItem.consolidateWith(eventId, content, tags)) {
-                        _consolidatedMeter.mark();
-                        continue;
-                    }
+                        // Check whether we've already added this piece of content to the poll result.  If so, consolidate
+                        // the two together to reduce the amount of work a client must do.  Note that the previous item
+                        // might be from a previous batch of events and it's possible that we have read two different
+                        // versions of the same item of content.  This will prefer the most recent.
+                        Item previousItem = uniqueItems.get(coord);
+                        if (previousItem != null && previousItem.consolidateWith(eventId, content, tags)) {
+                            _consolidatedMeter.mark();
+                            continue;
+                        }
 
-                    // If, due to "padding" we asked for too many events, release the claims for the next poll().
-                    if (items.size() == limit) {
-                        eventIdsToUnclaim.add(eventId);
-                        continue;
-                    }
+                        // If, due to "padding" we asked for too many events, release the claims for the next poll().
+                        if (items.size() == limit) {
+                            eventIdsToUnclaim.add(eventId);
+                            continue;
+                        }
 
-                    // We have found a new item of content to return!
-                    Item item = new Item(eventId, eventOrder.get(coord), content, tags);
-                    items.add(item);
-                    uniqueItems.put(coord, item);
+                        // We have found a new item of content to return!
+                        Item item = new Item(eventId, eventOrder.get(coord), content, tags);
+                        items.add(item);
+                        uniqueItems.put(coord, item);
+                    }
                 }
-            }
+            } while (rawEventIterator.hasNext() && stopwatch.elapsed(TimeUnit.MILLISECONDS) < MAX_POLL_TIME.getMillis());
```

Comment:
Should we temporarily log the subscription name if we encounter the time elapses on the polls? (I think this should be infrequent, so logging may be fine). I feel this may be an another way to alert us about any "large records" written badly by the writers.   
We may also add a metric meter as we did for the queue draining case.